best-practices/ml-platform/terraform/modules/node-pools/nodepools.tf

```diff
@@ -90,3 +91,31 @@ resource "google_container_node_pool" "node-pool" {
     update = "20m"
   }
 }
+
+# Create dedicated service account for node pool
+resource "google_service_account" "nodepool_sa" {
```

Comment:
Can you remove the `_sa` from the resource name as it is redundant based on the resource type.

https://cloud.google.com/docs/terraform/best-practices-for-terraform#naming-convention